packages/@aws-cdk/aws-synthetics/lib/code.ts

```diff
@@ -107,6 +106,8 @@ export class AssetCode extends Code {
       });
     }
 
+    this.validateCanaryAsset(scope, handler, family);
```

Comment:
I'm not sure if there was a specific reason for this ordering, but looks like this logic predates Asset's bundling support so there doesn't appear to be an issue doing this now.

Reply:
I think the ordering just has to do with the principle of throwing errors as early as possible.